src/nvim/ui.c

```diff
@@ -85,21 +86,25 @@ static char uilog_last_event[1024] = { 0 };
 //
 // See http://stackoverflow.com/a/11172679 for how it works.
 #ifdef _MSC_VER
-# define UI_CALL(funname, ...) \
+# define UI_CALL_IF(cond, funname, ...) \
     do { \
       UI_LOG(funname, 0); \
       for (size_t i = 0; i < ui_count; i++) { \
         UI *ui = uis[i]; \
-        UI_CALL_MORE(funname, __VA_ARGS__); \
+        if (cond) { \
+          UI_CALL_MORE(funname, __VA_ARGS__); \
```

Comment:
seems like `if (cond) {` should guard the whole thing. The logs would be confusing if `UI_LOG` writes a message even though `cond` skipped the actual work.

Reply:
The condition depends on the for loop, otherwise the macro would be completely pointless. We could show the LOG on the first time in the loop anything was called.